Xamarin.Forms.Platform.iOS/Renderers/ListViewRenderer.cs

```diff
@@ -364,14 +366,16 @@ void OnScrollToRequested(object sender, ScrollToRequestedEventArgs e)
 
 		void UpdateEstimatedRowHeight()
 		{
-			if (_estimatedRowHeight)
-				return;
-
 			var rowHeight = Element.RowHeight;
 			if (Element.HasUnevenRows && rowHeight == -1)
 			{
 				var source = _dataSource as UnevenListViewDataSource;
-				if (_shouldEstimateRowHeight)
+
+				// We want to make sure we reset the cached defined row heights whenever this is called.
+				// Failing to do this will regress Bugzilla 43313 (strange animation when adding rows with uneven heights)
+				source.CacheDefinedRowHeights();
```

Comment:
we are using source without checking if it's null.. not sure `?. `will be more correct